src/Microsoft.PowerShell.Commands.Utility/commands/utility/MatchString.cs

```diff
@@ -1518,6 +1600,9 @@ private bool DoMatchWorker(string operandString, MatchInfo matchInfo, out MatchI
             int patternIndex = 0;
             matchResult = null;
 
+            var indexes = new List<int>();
+            var lengths = new List<int>();
```

Comment:
I wonder why we need this if already have the same in matchResult.Matches?
Note that in both cases we have disordered list.

Reply:
I'm creating new lists in all cases, although I could use matchResult.Matches in most cases and allocate and use these lists only when the SimpleMatch parameter is enabled. If the SimpleMatch parameter is enabled, then the current behavior is that Matches does not contain any Match objects. Without any Match objects, I would need to pass in these lists to implement the colored text when SimpleMatch is enabled.

Reply:
I think we should avoid creating or populating lists in the (possibly common) case where `Emphasize` is not specified or where VT100 isn't available. We don't want a performance hit in cases where there's no benefit.

That might look something like:

```csharp
List<int> indexes;
List<int> lengths;

if (Emphasize)
{
    indexes = new List<int>();
    lengths = new List<int>();
}

...

if (Emphasize)
{
    foreach (Match match in matches)
    {
        indexes.Add(match.Index);
        lengths.Add(match.Length);
    }
}

// etc.

...
    matchResult = Emphasize
        ? new MatchInfo(indexes, lenghts)
        : new MatchInfo();

    matchResult.IgnoreCase = !CaseSensitive;
    Line = operandString;
    // etc.
...
```